### ISSUE TITLE
Add pygame.require_version() function

### ISSUE THREAD

**Issue by robertpfeiffer** [pygame/pygame#1738]
`pygame.require_version()` would be a function with one mandatory positional/keyword parameter for the minimal pygame version a game should run on, like this: `pygame.require_version(pygame="2.0.0-dev1")`

Optional keyword parameters could be SDL version, Python version, and OpenGL version like this: 

```
pygame.require_version(pygame="2.0.0-dev1",
                                        python="3.6",
                                        sdl="2.0.8",
                                        gl="3.3")
```

The goal/behaviour of this function would be to check for the required versions of dependencies, and if any of them are not met, to open a window prompting the user to update their dependencies, and then *terminate the program*. This function should not be called from imported modules, but only from directly from `__main__`, right after `pygame.init()`.

I want to leave the behaviour of `pygame.require_version(pygame="2.0.0")` in a hypothetical future pygame version 2.1 deliberately unspecified. Maybe it will display a window warning the user, like löve does, or maybe it will set some kind of compatibility flag.

It would be nice if  `pygame.require_version(pygame="2.0.0")` could un-set some kind of PyGame 1.X compatibility flag already, but I realise that this would make things confusing and possibly limit adoption of this feature. At the same time using  `pygame.require_version(pygame="1.9.6")` to *enable* compatibility features would be pointless, because this function didn't exist yet in PyGame 1.9.6.

If we decide to add a function like this, it would be a good idea to add it as soon as possible, or not at all.

**Related Docs**: https://www.pygame.org/docs/ref/pygame.html

**Comment by AaronLi** [comment]
It seems like getting pygame version, python version, and sdl version are easy enough though the builtin pygame.version doesn't support reporting the `-dev1` or `.dev8` there is pygame.ver to go off of. Would it be ok to split the version string on the `.` and `-` then continue from there?

**Comment by MyreMylar** [comment]
Eventually the dev numbers will go away when pygame 2 is officially released, so this is roughly what I've done (for now) in `pygame_gui`:

```
if 'dev' in pygame.ver.split('.')[-1]:
    PYGAME_DEV_NUM = int(re.findall(r'\d+', pygame.ver.split('.')[-1])[0])
else:
    PYGAME_DEV_NUM = 10

VERSION_GREATER_THAN_OR_EQUAL_2_DEV10 = pygame.version.vernum[0] >= 2 and PYGAME_DEV_NUM >= 10
``

Which should support the release version of pygame 2 and dev versions greater than or equal to 10.

**Comment by MyreMylar** [comment]
> So do you think this is a good idea or not?

I haven't personally found a need for a function that works in the way you describe yet, as my focus has largely been on making stuff that works on as many pygame versions as possible that is not surprising. For my library code (pygame_gui) I have, so far, mostly been building multiple code paths to support features on different versions. So I've been using similar version testing code - but not with the intent to shut down an app, just to branch paths.

For any games, I either haven't cared that much about what versions they work on because I was using them in education in a specific known environment - or figured I would eventually freeze them with a specific version of python built-in.

I can see how others might find it useful though, so I was reserving judgement. Probably it should go in the version module though.

**Comment by MightyJosip** [comment]
Should [MessageBox](https://wiki.libsdl.org/SDL_ShowMessageBox) (example [here](https://discourse.libsdl.org/uploads/default/original/2X/b/b1bf7f530880550c6cd308b117909dddddcd3d73.jpeg)) be the way to go to handle this.

Also what should be the result if the user calls this function in the middle of the game loop (what should happen to the original window, or we would enable this function only if no window is initialised (so between `pygame.init()` and `pygame.display.set_mode()`).

Or when we are mentioning this MessageBox should we just port the function from the SDL itself (because it can be useful in the case of missing image/sound/some other resource). Than the user would have to write his own require version, which would be something like this
```
if pygame.version.vernum <= "something":
    mb = pygame.display.show_message_box("Update PyGame", "Quit")
```

**Comment by illume** [comment]
tldr; it'd probably work better in a separate library IMHO.

The main problem with a pygame.require_version I see is that it doesn't really work for older versions of pygame. If you consider the install base now, mostly people have some 1.9.4 or 1.9.6 version of pygame, and then others have various 2.0.0.devx versions. So if you want to require that version, you'd need custom code. Probably it would work better as a separate library for this reason.

The message box code in solarwolf (and I think stuntcat) try and show a message box if there's an exception. Which is important if you launch from a GUI and there's no console for the user to see the message. One of the cases it handles is if pygame isn't installed - it shows a message box trying to use other gui toolkits. I mention this, because it's definitely useful. But it goes a step further than this because it can be used for catching exceptions other than ImportError.

Python has specified dependency management now, so probably it maybe a good idea to use the same ones that the packaging world uses (requirements.txt style dependencies)

**Comment by ankith26** [comment]
Stumbled across this while issue gardening, NGL I quite like the idea of the `pygame.require_version` and would have liked it to make it by pygame 2.0.0, but I also agree it works even better as part of a separate lib, as in that case it could work on older versions. Also, the apps might have to handle stuff differently than opening a messagebox, so it's not a bad idea for programmers using pygame to roll their own utils code to handle this the way they like :)
